BSP/Simulator/Hardware/Timer.py

```diff
@@ -0,0 +1,37 @@
+import csv
+import os
+
+# Path of file
+file = "Data/Timer.csv"
+
+def update():
+	""" Decrements the current value of the two timer simulations in Timer.csv every 1 ms """
+
+	# Create the file if it doesn't exist yet
+	os.makedirs(os.path.dirname(file), exist_ok=True)
+	if not os.path.exists(file):
+		open(file, 'w')
+
+	current_values = []
+	reload_values = []
+
+	with open(file, 'r') as csvfile:
+		reader = csv.reader(csvfile, delimiter=',')
+		
+		try:
+			for row in reader: # Should be 2 rows
+				current_values.append(int(row[0]))
+				reload_values.append(int(row[1]))
+
+		except IndexError: # If the csv is blank.
+			current_values.extend([0x100000, 0x1000000]) # 0xFFFFFF + 1 since we are decrementing below
+			reload_values.extend([0xFFFFFF, 0xFFFFFF])

```

Comment:
This try/except isn't actually catching anything. If the file is blank, then the `reader` object has now `row`, so the loop will not even execute. The IndexError is happening in the next block when you try to write to the file. That's where you should catch the exception.